framework/src/samplers/Sampler.C

```diff
@@ -12,103 +12,76 @@
-                    parameters.get<THREAD_ID>("_tid"),
-                    false),
+    SetupInterface(this),
     DistributionInterface(this),
-    Restartable(parameters, "Samplers"),
-    _tid(getParam<THREAD_ID>("_tid")),
-    _reseed_for_new_sample(getParam<bool>("reseed")),
-    _dist_names(getParam<std::vector<DistributionName>>("distributions")),
-    _var_names(getParam<std::vector<std::string>>("perturb_parameters")),
-    _current_sample(0),
-    _failed_runs(true)
-{
-  _var_dist_map.clear();
-  _probability_weight.clear();
-  _var_value_hist.clear();
-  if (_var_names.size() != _dist_names.size())
-    mooseError("The size of perturb_parameters (",
-               _var_names.size(),
-               ") != the size of distributions (",
-               _dist_names.size(),
-               ").");
-  if (!_var_names.empty())
-  {
-    for (unsigned int i = 0; i < _var_names.size(); ++i)
-    {
-      _var_dist_map[_var_names[i]] = &getDistributionByName(_dist_names[i]);
-      _var_value_map[_var_names[i]] = 0.0;
-      _var_value_hist[_var_names[i]] = std::vector<Real>();
-    }
-  }
-}
-
-void
-Sampler::generateSamples()
-{
-}
-
-std::vector<std::string>
-Sampler::getSampledVariableNames()
+    _distribution_names(getParam<std::vector<DistributionName>>("distributions")),
+    _seeds(getParam<std::vector<unsigned int>>("seeds"))
 {
-  return _var_names;
-}
+  for (std::size_t i = 0; i < _seeds.size(); ++i)
+    _generator.seed(i, _seeds[i]);
 
-std::vector<Real>
-Sampler::getSampledValues(const std::vector<std::string> & variableNames)
-{
-  std::vector<Real> vals;
-  for (auto & var_name : variableNames)
-  {
-    Real val = getSampledValue(var_name);
-    vals.push_back(val);
-  }
-  return vals;
+  for (const DistributionName & name : _distribution_names)
+    _distributions.push_back(&getDistributionByName(name));
 }
 
-Real
-Sampler::getSampledValue(const std::string & variableName)
+void
+Sampler::execute()
 {
-  std::map<std::string, Real>::iterator it;
-  it = _var_value_map.find(variableName);
-  if (it == _var_value_map.end())
-    mooseError(
-        "Could not find the parameter: ", variableName, " in the list of perturbed parameters!");
-  return it->second;
+  // Get the samples then save the state so that subsequent calls to getSamples returns the same
+  // random numbers until this execute command is called again.
+  getSamples();
+  _generator.saveState();
 }
 
-std::vector<Real>
-Sampler::getProbabilityWeights()
+std::vector<DenseMatrix<Real>>
+Sampler::getSamples()
 {
-  return _probability_weight;
+  _generator.restoreState();
+  sampleSetUp();
+  std::vector<DenseMatrix<Real>> output(_distributions.size());
+  for (std::size_t i = 0; i < _distributions.size(); ++i)
+    output[i] = sampleDistribution(*_distributions[i], i);
+  sampleTearDown();
+  return output;
 }
 
-bool
-Sampler::checkRuns()
+void
+Sampler::checkSeedNumber(unsigned int required) const
 {
-  return _failed_runs;
+  if (_seeds.size() < required)
```

Comment:
OK - so somebody does check for a certain number of seeds. This still seems like a really bad idea.